File: solution/data_structure/18258/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/9bc206242acc4800a1f1df61d2b1b062
#include<bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    int T; cin >> T;
    queue<int> Q;
    while(T--) {
        string cmd; cin >> cmd;
        if(cmd == "push") {
            int X; cin >> X;
            Q.push(X);
        }
        else if(cmd == "pop") {
            if(Q.empty()) cout << -1 << '\n';
            else {
                cout << Q.front() << '\n';
                Q.pop();
            }
        }
        else if(cmd == "size") {
            cout << (int)Q.size() << '\n';
        }
        else if(cmd == "empty") {
            cout << Q.empty() << '\n';
        }
        else if(cmd == "front") {
            if(Q.empty()) cout << -1 << '\n';
            else cout << Q.front() << '\n';
        }
        else if(cmd == "back") {
            if(Q.empty()) cout << -1 << '\n';
            else cout << Q.back() << '\n';
        }
    }

    return 0;
}

File: solution/data_structure/18258/main.py
# Authored by : gusdn3477
# Co-authored by : -
# Link : http://boj.kr/6bf4491116ff480fab750a65591c134e

import sys
from collections import deque
def input():
    return sys.stdin.readline().rstrip()

queue = deque()
N = int(input())
for i in range(N):
    com = input().split()
    if com[0] == 'push':
        queue.append(com[1])
    elif com[0] == 'pop':
        if queue:
            print(queue.popleft())
        else:
            print(-1)
    elif com[0] == 'size':
        print(len(queue))
    elif com[0] == 'front':
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif com[0] == 'back':
        if queue:
            print(queue[-1])
        else:
            print(-1)
    elif com[0] == 'empty':
        if not queue:
            print(1)
        else:
            print(0)